app/src/main/java/com/example/placefinder/MainActivity.kt

```diff
@@ -36,15 +36,15 @@ class MainActivity : ComponentActivity() {
         override fun onCreate(savedInstanceState: Bundle?) {
             super.onCreate(savedInstanceState)
             setContent {
-                viewModel.fetchParks()
-                val parks by viewModel.parks.observeAsState(initial = emptyList())
+                var parks by remember { mutableStateOf(emptyList<Address>()) }
```

Comment:
In the given code, viewModel.fetchParks() method is being called on the main thread. This method could take a considerable amount of time to complete, which could freeze the UI thread and make the app unresponsive. To avoid this, I moved this operation to a background thread.

Reply:
LaunchedEffect to run coroutines in compose... good one!